### ISSUE TITLE
rosserial_python can not handle ttyO1

### ISSUE THREAD

**Issue by FriedCircuits** [ros-drivers/rosserial#82]
I noticed that if the serial port is /dev/ttyO1 it returns error can not open port /dev/tty01.

It seems to not handle letter after tty.

I traced it down to SerialClient.py 

I think the bug it in there.

**Comment by FriedCircuits** [comment]
I mapped device name to /dev/tty80 and that worked.

So it definitely can't handle tty names with letters

**Comment by mikeferguson** [comment]
The only place the port name is used is in opening the port using PySerial, thus you either have a permissions error, or there is some underlying issue with your instance of PySerial (I tend to think it is permissions problems).

In a python interpreter, can you do the following to test your setup and sort out permissions on the port:

```
     from serial import Serial
     port = Serial("/dev/ttyO1", 57600)
```

**Comment by mikeferguson** [comment]
Actually, portnames are either /dev/ttyUSB\* or /dev/ttyACM\* for 99.9% of users, so clearly it does handle "letters after tty".

**Comment by FriedCircuits** [comment]
I meant letters after tty or USB. I ln -s the port to another name without letters worked. I don't think its permissions.

**Comment by FriedCircuits** [comment]
Doing that in python didn't return any errors

**Comment by FriedCircuits** [comment]
By this is on the BeagleBone Black.

**Comment by PaulBouchier** [comment]
I think you're describing a configuration where Beaglebone black is a rosserial client, connecting to a PC running ROS. Is that correct? So I guess you're using rosserial_embeddedlinux - right?

You should be opening the serial port in the following way:
nh.initnode("/dev/ttyO1");

IOW the initnode method supports opening without a parameter (suitable for Arduino, where there's one serial port and it's name is well known (Serial, or Serial1). That's inappropriate for many linux systems, so the initnode method supports naming the port to open, which can be an IP address or a name like  /dev/ttyXXX.

Does this help? Or do you still have a problem?

**Comment by FriedCircuits** [comment]
Thanks, the problem went away after the last reboot. But actually the BBB is the ROS host. And using ROSSerial python server with an Arduino. I got it working well last night with the Arduino Due and sending temp and humidity data. Just submitted a pull request for the recipes to cross compile ROSSerial for ARM earlier today
https://github.com/bmwcarit/meta-ros Everything compiles except for ROSSerial Server.

I never could get the Due or Leonardo to work over USB though. I submitted a issue for the Due since it wouldn't compile without a change to the ArduinoHardware.h.
